### DISCUSSION TITLE
PDF

### DISCUSSION THREAD

**Discussion by Oneupupup** [vuepress-theme-hope/vuepress-theme-hope#3451] [Q&A] [answered]
Set "plugin.components" in theme options to customize it.

我想导入 pdf 但是一直不知道在哪里设置，一直报这个警告

**Comment by fdevans** [comment, marked as answer]
Our works with the following config. (reduced size to focus on the important part)

```
  theme: hopeTheme({
    plugins: {
      components: {
          components: [
            "FontIcon",
            "PDF",
            "VideoPlayer",
            "YouTube",
          ],
          componentOptions: {
            fontIcon: {
              assets: "fontawesome",
            },
            pdf: {
              pdfjs: "/assets/lib/pdfjs/",
            },
          }
      }
    },
])
```

Then in the page use `<PDF url="/files/your.pdf" page="1"/>`

The full example here: https://github.com/rundeck/docs/blob/vuepress2-migration/docs/.vuepress/config.ts
